source/Engine.cpp

```diff
@@ -1633,6 +1639,36 @@ void Engine::MoveShip(const shared_ptr<Ship> &ship)
 
 
 
+// Changing the players flagship during flight requires an update of all active
+// data that references the players flagship.
+void Engine::ChangePlayerFlagship(const shared_ptr<Ship> oldFlagship, shared_ptr<Ship> &newFlagship)
+{
+	// Remove active data in the old flagship.
+	if(oldFlagship)
+	{
+		oldFlagship->SetTargetAsteroid(nullptr);
+		oldFlagship->SetTargetFlotsam(nullptr);
+		oldFlagship->SetTargetShip(nullptr);
+		oldFlagship->SetTargetStellar(nullptr);
+		oldFlagship->SetTargetSystem(nullptr);
+	}
+	
+	// Let PlayerInfo handle the default administration (like the players escorts).
+	player.SetFlagship(newFlagship);
+	
+	// Update all ships that still have the old flagship as parent, for example
+	// NPCs that are following the player.
+	for(std::shared_ptr<Ship>& ship : ships)
+	{
+		if(ship && ship->GetParent() == oldFlagship)
```

Comment:
I'm pretty sure that `ship` can never be `nullptr`.

Reply:
You're probably right, but I consider it good practice to still check the pointer; for just in case some future code does cause the pointer to become `nullptr` (for example because it decides to prune destroyed ships more actively in a multi-threaded environment).